osquery/tables/system/windows/logical_drives.cpp

```diff
@@ -16,53 +16,27 @@ QueryData genLogicalDrives(QueryContext& context) {
   QueryData results;
 
   const WmiRequest wmiLogicalDiskReq(
-      "select DeviceID, DriveType, FreeSpace, Size, FileSystem from "
+      "select DeviceID, Description, FreeSpace, Size, FileSystem from "
       "Win32_LogicalDisk");
-  const std::vector<WmiResultItem>& wmiResults = wmiLogicalDiskReq.results();
-  for (unsigned int i = 0; i < wmiResults.size(); ++i) {
+  auto const& logicalDisks = wmiLogicalDiskReq.results();
+  for (const auto& logicalDisk : logicalDisks) {
     Row r;
-    unsigned int driveType = 0;
     std::string deviceId;
-    wmiResults[i].GetString("DeviceID", deviceId);
-    wmiResults[i].GetUnsignedInt32("DriveType", driveType);
-    wmiResults[i].GetString("FreeSpace", r["free_space"]);
-    wmiResults[i].GetString("Size", r["size"]);
-    wmiResults[i].GetString("FileSystem", r["file_system"]);
+    logicalDisk.GetString("DeviceID", deviceId);
+    logicalDisk.GetString("Description", r["type"]);
```

Comment:
I'm totally noob here.  I just found this thing https://docs.microsoft.com/en-us/windows/desktop/cimwin32prov/win32-logicaldisk . But it does not say anything about the expected values of the Description field.
Can we guarantee that type column will be exactly the same as it was before? We don't normally change column behavior.

Reply:
The `Description` field has a `DisplayName` CIM qualifier, meaning that it provides the canonical string representation of the object.

`DriveType` is almost identical, but it just returns one of the values from `GetDriveTypeA` instead (which we then have to switch on manually). I'd argue that the current approach is actually more incorrect, since:

1.  It doesn't use the canonical string representations (e.g. "Compact Disc" instead of "CD-ROM Disk")
2. It will need to be updated each time Windows adds a new logical disk type

`Description` has neither of these problems. It's also inherited from `CIM_ManagedSystemElement`, making it pretty unlikely that Microsoft will ever break its current behavior.

Reply:
I see your point and totally agree that canonical representation is better to have/use.

However, we can not just change column because it's better. People use it the way it is.
What I would suggest, is to add new column let's say canonical_type and leave the type as is.

What do you think?

Reply:
> However, we can not just change column because it's better. People use it the way it is.
What I would suggest, is to add new column let's say canonical_type and leave the type as is.

I think this would overly complicate the table: we would have two nearly-identical columns with no clear guidance on which column is actually the *correct* one to rely on. That would confuse me as an end user, and sets a bad precedent for future fixes.

It's also worth noting that the current `type` column is unreliable -- I occasionally get all "Unknown"s with it, per the original PR (#5367).

As far as I can tell, osquery doesn't use semantic versioning **or** have a formal deprecation policy. As such, I don't think we should let the perfect (backwards compatibility) be the enemy of the good (fixing defects) here.

Reply:
For reference, here is the current (`experimental` branch) behavior of the table on my machine:

```
osquery> select * from logical_drives;
+-----------+---------+-------------+-------------+-------------+----------------+
| device_id | type    | free_space  | size        | file_system | boot_partition |
+-----------+---------+-------------+-------------+-------------+----------------+
| C:        | Unknown | 18973102080 | 53109321728 | NTFS        | 0              |
| D:        | Unknown | 0           | 58013696    | CDFS        | 0              |
+-----------+---------+-------------+-------------+-------------+----------------+
```

Note that `type` is *totally* incorrect at the moment. As such, I don't see a serious backwards compatibility issue with these fixes.

Reply:
Leaving old columns as is, because of the backward compatibility is definitely ugly and confusing. However, people can not check updates of osquery daily, and update queries accordingly. 
The only solution I see is to change type column description to something like "Deprecated key, replaced by canonical_type". That way, we will partially deal with the end user confusion at least.